lib/install/config/loaders/core/assets.js

```diff
@@ -1,11 +1,21 @@
-const { env, publicPath } = require('../configuration.js')
+const Url = require('url-parse')
+const { env, paths, publicPath } = require('../configuration.js')
+
+// Compute path for internal pack assets if ASSET_HOST is supplied
+const computeAssetPath = (host) => {
+  if (!host) { return `/${paths.entry}/` }
+  const { slashes, href } = new Url(host)
+  if (slashes) { return `${href}/${paths.entry}/` }
+  if (href) { return `//${href}/${paths.entry}/` }
+  return `/${paths.entry}/`
```

Comment:
This is a bit confusing flow. Can we replace with a case statement instead? Not a big fan of multiple exit functions like this.